src/Mvc/Mvc.Testing/src/WebApplicationFactory.cs

```diff
@@ -136,10 +159,92 @@ internal virtual WebApplicationFactory<TEntryPoint> WithWebHostBuilderCore(Actio
         return factory;
     }
 
-    [MemberNotNull(nameof(_server))]
-    private void EnsureServer()
+    /// <summary>
+    /// Configures the factory to use Kestrel as the server.
+    /// </summary>
+    public void UseKestrel()
     {
-        if (_server != null)
+        if (_server != null || _webHost != null)
```

Comment:
```suggestion
        if (ServerStartedl)
```

I think this should also throw if `_host` is set.

Reply:
Thanks, I had missed this one.